sdk/provisioning/Azure.Provisioning.CloudMachine/src/AzureSdkExtensions/OpenAIModel.cs

```diff
@@ -22,10 +19,28 @@ public OpenAIModel(string model, string modelVersion, AIModelKind kind = AIModel
 
     public string Model { get; }
     public string ModelVersion { get; }
-    private AIModelKind Kind { get; } // has to be $"{cloudMachine.Id}_embedding" or {cm.Id}
-    public CognitiveServicesAccount? CognitiveServices { get; set; } = default;
+    private AIModelKind Kind { get; }
+
+    internal OpenAIFeature Account { get; set; } = default!;
+
+    private OpenAIFeature GetOrCreateOpenAI(CloudMachineInfrastructure cm)
+    {
+        foreach (OpenAIFeature feature in cm.Features.FindAll<OpenAIFeature>())
+        {
+            return feature;
+        }
+        var openAI = new OpenAIFeature();
+        cm.AddFeature(openAI);
+        return openAI;
+    }
 
     public override void AddTo(CloudMachineInfrastructure cm)
+    {
+        OpenAIFeature openAI = GetOrCreateOpenAI(cm);
```

Comment:
Is it intentional that this will only return the first feature found even though there could be multiple?

Reply:
Yeah, it's intentional. A model cannot be associated with more than one account. I was thinking about detecting this conditional and throwing, but I am not sure it's worth the perf hit.